### PR TITLE
save the form data in the local storage of the browser

### PR DESCRIPTION
**Pull request by khitermedachraf**
**In this Pull Request, we have implemented the following features:**

1. When the user changes the content of any input field, the data is saved to the local storage.
2. When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.
3.  create a single JavaScript object with all the data from the entire form and save it in local storage

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -121,4 +121,67 @@ form.addEventListener('submit', (event) => {
...
+if (storageAvailable('localStorage')) {
+  // Yippee! We can use localStorage awesomeness
+  // Testing whether your storage has been populated
+  if (!localStorage.getItem('inputsData')) {
+    populateStorage();
+  } else {
+    setInputsData();
+  }
+}
+
```
Comment (sja-thedude) [inline app.js:179]: - [x] Kindly make sure to write code for `when the user loads the page, any saved data in the local storage have input fields with pre-filled data`. 
I have attached a video where when we refresh the page the data on the input files should be persistent. You have to use `window.onload ` for this. Kindly work on it.

https://user-images.githubusercontent.com/70269507/162326319-3d65e196-b835-40be-b5ed-7255b344d1c4.mov

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @khitermedachraf, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Good job in implementing the local storage
✔️ Good PR Title and description

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js (hunk 1 above)
Comment (khitermedachraf) [reply app.js:179]: Hi @sja-thedudem, I hope you are doing well,
**I notice that the event listener I used for this activity (`''change"`)** works fine when the input loses focus, so I used also the **`window.onload`**  but it behaves just one the same manner, I think I should use the `"input"` event listener to **capture the changes instantly.**

**Comment by khitermedachraf** [comment]
Hi @sja-thedude, I hope you are doing well,
**I notice that the event listener I used for this activity `("change")`** works fine when the input loses focus, so I used also the **`window.onload`**  but it behaves just in the same manner, I think I should use the `"input"` event listener to **capture the changes instantly.**@

File: app.js (hunk 1 above)
Comment (khitermedachraf) [reply app.js:179]: ### Done with "input" event listener.

**Review by micheaol — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hi @khitermedachraf , 

Great work on making the changes requested by the previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

well-1

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**